Wrappers/Python/cil/framework/framework.py

```diff
@@ -3351,20 +3357,167 @@ def dot(self, other, *args, **kwargs):
+            
+            new_dimensions = numpy.array(self.dimension_labels)
+            new_dimensions = tuple(numpy.delete(new_dimensions, kwargs['axis']))
+            if result.shape == out.shape and new_dimensions == out.dimension_labels:
+                out.fill(result.astype(out.dtype)) # convert to the type given in out
+            else:
+                raise ValueError('Data mismatch: out.shape = {} result.shape = {}, out.dimension_labels = {} result.dimension_labels = {}'.format(out.shape, result.shape, out.dimension_labels, new_dimensions))   
+
+        elif issubclass(type(out), numpy.ndarray):
+            if direction is not None:
+                kwargs['axis'] = self.get_dimension_axis(direction)
+            pwop(self.as_array(), *args, **kwargs)
+
+        else:
+            raise ValueError (message(type(self),  "Incompatible class:" , pwop.__name__, type(out)))
+
+    def sum(self, direction=None, *args, **kwargs):
+        """
+        Returns the sum of values in the DataContainer
+        
+        Parameters
+        ----------
+        direction : string or tuple of strings, optional
+            Specify the axis or axes to calculate the sum along using a dimension_label. 
+            Default is None, calculates the sum of the whole array
+        out : ndarray or DataContainer, optional
+            Provide an object in which to place the result. The object must have the correct dimensions and 
+            (for DataContainers) the correct dimension_labels, but the type will be cast if necessary.
+            Default is None
+        
+        Returns
+        -------
+        scalar or DataContainer
+            The sum as a scalar or inside a DataContainer with reduced dimension_labels
+            Default is to accumulate and return data as float64 or complex128
+        """     
+        if kwargs.get('dtype', None) is not None:
+            logging.WARNING("dtype argument is ignored, using float64 or complex128")
+        
+        if numpy.isrealobj(self.array):
             kwargs['dtype'] = numpy.float64
-        return numpy.mean(self.as_array(), *args, **kwargs)
+        else:
+            kwargs['dtype'] = numpy.complex128
+
+        return self._directional_reduction_unary(numpy.sum, direction=direction, *args, **kwargs)
+
+    def min(self, direction=None, *args, **kwargs):
+        """
+        Returns the min pixel value in the DataContainer
+        
+        Parameters
+        ----------
+        direction : string or tuple of strings, optional
+            Specify the axis or axes to calculate the minimum along using a dimension_label.
+            Default is None, calculates the min of the whole array
+        out : ndarray or DataContainer, optional
+            Provide an object in which to place the result. The object must have the correct dimensions and 
+            (for DataContainers) the correct dimension_labels, but the type will be cast if necessary.
```

Comment:
Since it is numpy casting the result, why not add the additional text here and in the other reductions.

```suggestion
            (for DataContainers) the correct dimension_labels, but the type will be cast if necessary. See [Output type determination](https://numpy.org/doc/stable/user/basics.ufuncs.html#ufuncs-output-type) for more details.
```